pkg/operation/botanist/component/nginxingress/nginxingress.go

```diff
@@ -0,0 +1,616 @@
+		vpa        = &autoscalingv1beta2.VerticalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      controllerName,
+				Namespace: n.namespace,
+			},
+			Spec: autoscalingv1beta2.VerticalPodAutoscalerSpec{
+				TargetRef: &autoscalingv1.CrossVersionObjectReference{
+					APIVersion: appsv1.SchemeGroupVersion.String(),
+					Kind:       "Deployment",
+					Name:       deploymentController.Name,
+				},
+				UpdatePolicy: &autoscalingv1beta2.PodUpdatePolicy{
+					UpdateMode: &updateMode,
+				},
+				ResourcePolicy: &autoscalingv1beta2.PodResourcePolicy{
+					ContainerPolicies: []autoscalingv1beta2.ContainerResourcePolicy{
+						{
+							ContainerName: autoscalingv1beta2.DefaultContainerResourcePolicy,
+							MinAllowed: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("25m"),
+								corev1.ResourceMemory: resource.MustParse("100Mi"),
+							},
+						},
+					},
+				},
+			},
+		}
+
+		ingressClass *networkingv1.IngressClass
+	)
+
+	if version.ConstraintK8sGreaterEqual122.Check(n.values.KubernetesVersion) {
+		ingressClass = &networkingv1.IngressClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   n.values.IngressClass,
+				Labels: getLabels(labelValueController, ""),
+			},
+			Spec: networkingv1.IngressClassSpec{
+				Controller: "k8s.io/" + n.values.IngressClass,
+			},
+		}
+	}
+
+	return registry.AddAllAndSerialize(
+		clusterRole,
+		clusterRoleBinding,
+		serviceAccount,
+		configMap,
+		serviceController,
+		deploymentController,
+		podDisruptionBudgetController,
+		vpa,
+		roleBackend,
+		roleBindingBackend,
+		serviceBackend,
+		deploymentBackend,
+		ingressClass,
+	)
+}
+
+func getLabels(componentLabelValue string, releaseLabelValue string) map[string]string {
+	labels := map[string]string{
+		v1beta1constants.LabelApp: labelAppValue,
+		labelKeyComponent:         componentLabelValue,
+	}
+	if releaseLabelValue != "" {
+		labels[labelKeyRelease] = releaseLabelValue
+	}
+	return labels
+}
+
+func (n *nginxIngress) getArgs(configMap *corev1.ConfigMap) []string {
+	out := []string{
+		"/nginx-ingress-controller",
+		"--default-backend-service=" + n.namespace + "/" + serviceNameBackend,
+		"--enable-ssl-passthrough=true",
+		"--publish-service=" + n.namespace + "/" + serviceNameController,
+		"--election-id=ingress-controller-seed-leader",
+		"--update-status=true",
+		"--annotations-prefix=nginx.ingress.kubernetes.io",
+		"--configmap=" + n.namespace + "/" + configMap.Name,
+		"--ingress-class=" + n.values.IngressClass,
+	}
+	if version.ConstraintK8sGreaterEqual122.Check(n.values.KubernetesVersion) {
+		out = append(out, "--controller-class=k8s.io/"+n.values.IngressClass)
+	}
+	return out
+}
+
+// InterfaceMapToStringMap converts a map[string]interface{} to map[string]string
+func InterfaceMapToStringMap(in map[string]interface{}) map[string]string {
+	m := make(map[string]string, len(in))
+	for k, v := range in {
+		m[k] = fmt.Sprint(v)
+	}
+	return m
+}
```

Comment:
This function should probably be moved to the `pkg/operation/seed` package. This component can just take `map[string]string` as input

Reply:
addressed in gardener@65735a2